src/main/java/executor/service/service/impl/ExecutionServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package executor.service.service.impl;
+
+import executor.service.model.ProxyConfigHolder;
+import executor.service.model.Scenario;
+import executor.service.model.WebDriverConfig;
+import executor.service.service.ExecutionService;
+import executor.service.service.ScenarioExecutor;
+import executor.service.service.WebDriverInitializer;
+import org.openqa.selenium.WebDriver;
+
+/**
+ * The {@code ExecutionServiceImpl} class implements the {@link ExecutionService} interface and
+ * is responsible for executing scenarios using a WebDriver.
+ * It utilizes a {@link ScenarioExecutor} to perform the execution and relies on {@link WebDriverConfig} and
+ * {@link WebDriverInitializer} for configuring and initializing WebDriver instances.
+ * <p>
+ *
+ * @author Oleksandr Tuleninov
+ * @version 01
+ * @see executor.service.service.ScenarioExecutor
+ * @see executor.service.model.WebDriverConfig
+ * @see executor.service.service.WebDriverInitializer
+ */
+public class ExecutionServiceImpl implements ExecutionService {
+
+    private final ScenarioExecutor scenarioExecutor;
+    private final WebDriverConfig webDriverConfig;
+    private final WebDriverInitializer webDriverInitializer;
```

Comment:
Will it better to hide *WebDriverConfig* into *WebDriverInitializer*?
As I see from here it just enters into *webDriverInitializer.getInstance* with no additional logic to perform.